### ISSUE TITLE
Cursive doesn't use S3-wagon-private when downloading dependencies

### ISSUE THREAD

**Issue by danielcompton** [cursive-ide/cursive#1064]
We're using [s3-wagon-private](https://github.com/technomancy/s3-wagon-private) to host private dependencies. Cursive doesn't seem to use this wagon when it is downloading dependencies so we get errors saying the dependencies are missing. Running `lein deps` from the CLI downloads them to the local m2 cache so we can keep working, but it would be nice for it to all work from Cursive.

**Comment by rafaeldff** [comment]
We run into this all the time, multiple times per day. It manifests itself as an Event Log error of the sort:

```
2:17:55 PM Leiningen aborted: exit code 1: Could not transfer artifact thelib:thelib:jar:version from/to ourrepo (s3p://our-s3-repository): No connector available to access repository ourrepo (s3p://our-s3-repository) of type default using the available factories FileRepositoryConnectorFactory, WagonRepositoryConnectorFactory
```

**Comment by timgilbert** [comment]
I'm getting a similar problem using [lein-essthree](https://github.com/dparis/lein-essthree) with IntelliJ Community 2014.2.4 and  Cursive 1.4.0-eap3-2016.2, although the error message I'm seeing looks different:

```
Error synchronising lunar:0.1.0-SNAPSHOT: Failed to collect dependencies for [#object[org.sonatype.aether.graph.Dependency 0x46a17fba "org.clojure:clojure:jar:1.8.0 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x3a303e8e "org.clojure:clojurescript:jar:1.9.225 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x77a76f79 "reagent:reagent:jar:0.6.0-rc (compile)"] #object[org.sonatype.aether.graph.Dependency 0x632b1c88 "re-frame:re-frame:jar:0.8.0 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x2e3af3f8 "day8.re-frame:async-flow-fx:jar:0.0.6 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x15456cff "binaryage:devtools:jar:0.8.1 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x59551589 "prismatic:dommy:jar:1.1.0 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x62219d77 "bidi:bidi:jar:2.0.9 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x1bfd07f8 "kibu:pushy:jar:0.3.6 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x25e1b2ec "com.cemerick:url:jar:0.1.1 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x54691284 "cljs-http:cljs-http:jar:0.1.41 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x3acdf75b "com.workframe:paper:jar:0.5.0 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x49fcf53d "shodan:shodan:jar:0.4.2 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x61a2483c "com.taoensso:sente:jar:1.10.0 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x7d439bbe "lein-doo:lein-doo:jar:0.1.7 (test)"] #object[org.sonatype.aether.graph.Dependency 0x2861dc62 "org.clojure:tools.nrepl:jar:0.2.12 (compile)"] #object[org.sonatype.aether.graph.Dependency 0x61a814e "clojure-complete:clojure-complete:jar:0.2.4 (compile)"]]
```

**Comment by favila** [comment]
I use s3-wagon-private also, and do not have this problem. Cursive downloads dependencies from s3-based repos just fine.

**Comment by timgilbert** [comment]
With IntelliJ 2016.3.5 and Cursive 1.5.0-eap3-2016.3 the situation is changed but still doesn't work. In the "Maven / Repositories" section, the S3 repositories show up as "Error" when I try to index them, with the tooltip indicating `java.io.IOException: Wagon exception connecting to repository essthree-releases|s3://my-bucket/maven/releases/.index`.

In an effort to fix this I generated a maven index using https://github.com/apache/maven-indexer version 5.1.2-4333789 and uploaded it to the spot that is indicated in the exception above, but I still see the same error in the repositories tab.

**Comment by cursive-ide** [comment]
@timgilbert That is a different problem. Unfortunately the indexing is performed by IntelliJ, which doesn't understand the s3 wagon URLs. So as far as I can tell, there's no way to index these repositories, but the dependencies appear to be downloaded correctly. I've confirmed with @danielcompton that this now works for him too, so I'm going to close this issue. I've created #1686 to track the indexing issue.
